adapters/saml/core/src/main/java/org/keycloak/adapters/saml/rotation/SamlDescriptorPublicKeyLocator.java

```diff
@@ -179,7 +180,10 @@ private synchronized <T> Key refreshCertificateCacheAndGet(T key, Map<T, Key> ca
                 this.publicKeyCacheByKey.put(new KeyHash(x509certificate.getPublicKey()), x509certificate.getPublicKey());
             } else {
                 final X500Principal principal = x509certificate.getSubjectX500Principal();
-                String name = (principal == null ? "unnamed" : principal.getName()) + "@" + x509certificate.getSerialNumber() + "$" + UUID.randomUUID();
+                SecureRandom random = new SecureRandom();
+                byte[] bytes = new byte[16];
+                random.nextBytes(bytes);
+                String name = (principal == null ? "unnamed" : principal.getName()) + "@" + x509certificate.getSerialNumber() + "$" + Arrays.toString(bytes);
```

Comment:
Using `Arrays.toString` would create something like `[56, -120,...]` which is a horrible id IMHO. Why don't you create a new utility method in the `SecretGenerator` class to generate the random array and return it in hexa?

```java
    public static String randomBytesHexa(int length) {
        final StringBuilder sb = new StringBuilder();
        for (byte b : randomBytes(length) {
            sb.append(Character.forDigit((b >> 4) & 0xF, 16));
            sb.append(Character.forDigit((b & 0xF), 16));
        }
        return sb.toString();
    }
```

If you prefer to make it more similar to the UUID format you can also add the hyphen `-` after 4 or 8 bytes.